src/overlays/actors/ovl_En_Encount3/z_en_encount3.c

```diff
@@ -32,22 +34,173 @@ ActorInit En_Encount3_InitVars = {
     (ActorFunc)EnEncount3_Draw,
 };
 
-#endif
-
-extern UNK_TYPE D_060009A0;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD058.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD084.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD194.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD1EC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Draw.s")
+s32 D_809AD810 = false;
+
+void EnEncount3_Init(Actor* thisx, PlayState* play) {
+    EnEncount3* this = THIS;
+
+    this->unk14A = ENCOUNT3_GET_SPAWN_INDEX(thisx);
+    this->childParams = ENCOUNT3_GET_PARAM_F80(thisx);
+    this->switchFlags = ENCOUNT3_GET_SWITCHFLAGS(thisx);
+    this->unk15A = this->actor.csId;
+    this->unk16C = ENCOUNT3_GET_ROTZ(thisx) * 10.0f;
+    if (this->unk16C < 10.0f) {
+        this->unk16C = 10.0f;
+    } else if (this->unk16C > 1000.0f) {
+        this->unk16C = 1000.0f;
+    }
```

Comment:
```suggestion
    this->unk16C = CLAMP(this->unk16C, 10.0f, 1000.0f);
```
This might be a clamp macro, locking the min and max values the value can be.

Reply:
its not working

Reply:
Alright no worries.